File: app/src/main/java/bj/drunkmessengerblocker/main/MainComponent.java
package bj.drunkmessengerblocker.main;

import bj.drunkmessengerblocker.ActivityScope;
import bj.drunkmessengerblocker.AppComponent;
import dagger.Component;

/**
 * Created by j on 15/01/2017.
 */

@ActivityScope
@Component(dependencies = {AppComponent.class}, modules = {MainModule.class})
public interface MainComponent
{
    void inject(MainActivity mainActivity);
}

File: app/src/main/java/bj/drunkmessengerblocker/main/MainComponent.kt
package bj.drunkmessengerblocker.main

import bj.drunkmessengerblocker.ActivityScope
import bj.drunkmessengerblocker.AppComponent
import dagger.Component

/**
 * Created by Josh Laird on 23/05/2017.
 */
@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(MainModule::class))
interface MainComponent {
    fun inject(activity: MainActivity)
}